shared/AppInsightsCommon/Tests/Unit/src/ThrottleMgr.tests.ts

```diff
@@ -0,0 +1,642 @@
+import { Assert, AITestClass } from "@microsoft/ai-test-framework";
+import { AppInsightsCore, IAppInsightsCore, _eInternalMessageId } from "@microsoft/applicationinsights-core-js";
+import { IThrottleMsgKey } from "../../../src/Enums";
+import { IThrottleInterval, IThrottleLimit, IthrottleMgrConfig, IthrottleResult } from "../../../src/Interfaces/IThrottleMgr";
+import { ThrottleMgr} from "../../../src/ThrottleMgr";
+import { SinonSpy } from "sinon";
+import { Util } from "../../../src/Util";
+
+const compareDates = (date1: Date, date: string | Date) => {
+    try {
+        if (date1 && date) {
+            let date2 = typeof date == "string"? new Date(date) : date;
+            return date1.getUTCFullYear() === date2.getUTCFullYear() &&
+            date1.getUTCMonth() === date2.getUTCMonth() &&
+            date1.getUTCDate() === date2.getUTCDate();
+        }
+    } catch (e) {
+        console.log("compare dates error" + e);
```

Comment:
suggestion: As a test helper perhaps pass a additional "expected" match value and this this can be changed into an assert with the expected value "true/false" so if it's not as expected it would cause the test to fail.